streamalert/scheduled_queries/query_packs/manager.py

```diff
@@ -0,0 +1,312 @@
+            self._query_execution_id
+        )
+        return self._query_execution
+
+    def fetch_results(self):
+        """Refreshes the query_result property of this query pack
+
+        Returns:
+            AthenaQueryResult
+        """
+        if not self._query_execution.is_succeeded():
+            return None
+
+        self._query_result = self._execution_context.athena_client.get_query_result(
+            self._query_execution
+        )
+        return self._query_result
+
+    def save_to_cache(self):
+        entry = {
+            'query_execution_id': self._query_execution_id,
+            # 'query_string': self.generate_query_string(),
+        }
+
+        self._execution_context.state_manager.set(self.unique_id, entry)
+
+    def generate_query_string(self):
+        params = self._query_parameters
+        self._execution_context.logger.debug(
+            'Generated Parameters: {}'.format(json.dumps(params, indent=2))
+        )
+        return self._configuration.generate_query(**params)
+
+
+class QueryPacksManagerFactory:
+    """A factory service for generating QueryPacksManager instances"""
+
+    def __init__(self, execution_context):
+        self._execution_context = execution_context  # type: QueryPackExecutionContext
+
+    def new_manager(self):
+        """
+        Return:
+             QueryPacksManager
+        """
+        manager = QueryPacksManager(self._execution_context)
+        manager.load_query_configurations()
+
+        return manager
+
+
+class QueryPacksManager:
+    """This class manages multiple query packs that are firing off simultaneously
+
+    This class is not a service--it is a stateful container for QueryPacks, which themselves
+    can be stateful.
+    """
+
+    def __init__(self, execution_context):
+        self._execution_context = execution_context
+
+        self._query_configs = []  # type: list[QueryPackConfiguration]
+
+        self._query_packs = []  # type: list[QueryPack]
+
+    def load_query_configurations(self):
+        repo_packs = self._execution_context.query_pack_repository.get_packs()
+
+        # Tags are an inclusive filter
+        # If no tags are provided, then it includes all packs
+        # If multiple tags are provided, then only the packs that contain ALL OF THE TAGS
+        # will be run
+        configured_tags = self._execution_context.state_manager.get(
+            'streamquery_configuration', {}
+        ).get('tags', [])
+
+        for tag in configured_tags:
+            repo_packs = [
+                pack
+                for pack in repo_packs
+                if tag in pack.tags
+            ]
+
+        self._query_configs = repo_packs
+
+    def initialize_query_packs(self):
+        """Sets up query packs for this manager.
+
+        QueryPacks are a list of queries that this manager is intended to manage.
+        """
+
+        self._query_packs = []
```

Comment:
I think back in the internal days, because StreamQuery was designed to run off of lambda, bash, or `launchd`, the whole `nonblocking_single_pass()` could be executed in a single Python process OR on separate python processes OR on separate containers. This did some cleanup to ensure query packs were properly reloaded (in the case of `launchd`). I can get rid of it and merge it into the constructor